crates/e2e/tests/e2e/unwrap_cost_optimization.rs

```diff
@@ -0,0 +1,388 @@
+use crate::deploy::Contracts;
```

Comment:
Looks like the content of this file was moved from elsewhere. If there is any new code here it's difficult to tell....

Reply:
I started with the eth_integration tests as the basis for this new test but had to rewrite quite a bit.